File: test/matlab_src/idx2rad.m
% IDX2RAD Transform TOMS REFL matrix indices to radians.
%
% [theta,phi] = idx2rad(i,j,sy,sx)
%
% $Id: idx2rad.m,v 1.5 2006/05/17 14:39:18 danji Exp $

function [theta,phi] = idx2rad(i,j,sy,sx);
    sy = double(sy);
    sx = double(sx);
    CONST.d2r = pi/180;
    dx = 2*pi/sx;
    dy = pi/sy;
    phi = pi-dy/2-(double(i)-1)*dy;
    theta = (double(j)-1)*dx-pi+dx/2;
return

File: src/idx2rad.jl
# [scr/idx2rad.jl] 

""" 
Transforms from TOMS REFL Matrix indices to spherical coordinates (radians)

Arguments:
- i:  TOMS REFL Matrix latitude index of desired cell       |  Int
- j:  TOMS REFL Matrix longitude index of desired cell      |  Int 
- sy:  Number of latitude cells in grid                     |  Int 
- sx:  Number of longitude cells in grid                    |  Int 

Returns: (Tuple)
- (θ, ϵ):  Tuple of (Polar, Elevation) angle corresponding 
                to center of desired cell  (rad)             |  Tuple{Float, Float}
"""
function idx2rad(i::Int, j::Int, sy::Int, sx::Int)::Tuple{Float64, Float64}

    dx = 2 * pi / sx;
    dy = pi / sy;

    ϵ = pi - dy/2 - (i-1)*dy;
    θ = (j-1) * dx - pi + dx/2;

    return θ, ϵ
end